MRN-Code/fs_cronjobs
# Feature/verify data integrity

## Asana
https://app.asana.com/0/8755179914462/381292620508147

## Request/Problem
UNMC noticed that there were fewer dicom files on neptune than on their scanner. This has raised concerns about the mismatch of the number of files that are sent to MRN vs the number locally. 

## Solution
1. create a tool to allow users to enter the expected number of DICOM files per series on the Dicom QC tool.
2. Set up email notification to send to study owner if the number of files on archive do not match the expected number of files for the series.

## Test
- Make a copy of dbmap.json from devcoin6:/coins/config to your dev server.
- Go to /coins/crons, make sure the `$COINS_ENV === development`, run `php verifyDataIntegrity`
- After the command finished, an email should be sent to `nidev@mrn.org`
- The email should contain at least one series which indicates the mismatch between the actual number of files in archive and the expected number of file for such series.
- Test in production env: changing your dev machine environment to production by running `export $COINS_ENV=production`, changing the study email `tony.w.wilson@gmail.com` to YOUR email address.
- Run `php verifyDataIntegrity` again, now the email should send to both YOUR email and nidev.
- After finish testing, changing YOUR email back to `tony.w.wilson@gmail.com` and set your dev environment back to development `export $COINS_ENV=development`.
